neo/io/nixio.py

```diff
@@ -1190,14 +1190,15 @@ def _nix_attr_to_neo(nix_obj):
                         values = ""
                 elif len(values) == 1:
                     values = values[0]
-                elif prop.definition == ARRAYANNOTATION:
+                else:
+                    values = list(values)
```

Comment:
This is probably the reason the tests are failing.  The repetition of the `list()` conversion here is undoing the quantity creation from earlier in the method and the unit information is lost.  I think this `else` block is unnecessary.